### DISCUSSION TITLE
Use data from receiver webhook payload?

### DISCUSSION THREAD

**Discussion by hgibsonqb** [fluxcd/flux2#4377] [Q&A]
Hi,

I'd like to have a github webhook receiver that triggers a kustomization. Is there a way to extract the payload data from the receiver and pass it to the kustomization for the app to use?

**Comment by makkes** [comment]
The only thing that a Receiver does is trigger reconciliation of the resources declared in `.spec.resources`. All data necessary to properly reconcile a Kustomization should come from Git. So in your case you might want to put an intermediate step in between that extracts data from the webhook event and puts that data into Git, later to be consumed by Flux.

**Reply by hgibsonqb** [reply]
What would that intermediate step look like? Another application that receives webhooks from git, then a custom wehbook from that application to flux?

**Reply by hgibsonqb** [reply]
Git gives a lot of information in the webhook payload which the receiver in theory gets, but there's no way to see it?
